crates/vm/levm/src/hooks/default_hook.rs

```diff
@@ -307,92 +131,357 @@ impl Hook for DefaultHook {
+        .checked_mul(priority_fee_per_gas)
+        .ok_or(VMError::BalanceOverflow)?;
+
+    vm.increase_account_balance(vm.env.coinbase, coinbase_fee)
+}
+
+// In Cancun the only addresses destroyed are contracts created in this transaction
+pub fn delete_self_destruct_accounts(vm: &mut VM<'_>) -> Result<(), VMError> {
+    let selfdestruct_set = vm.accrued_substate.selfdestruct_set.clone();
+    for address in selfdestruct_set {
+        let account_to_remove = vm.get_account_mut(address)?;
+        *account_to_remove = Account::default();
+    }
+    Ok(())
+}
+
+pub fn validate_min_gas_limit(vm: &mut VM<'_>) -> Result<(), VMError> {
+    // check for gas limit is grater or equal than the minimum required
+    let calldata = vm.current_call_frame()?.calldata.clone();
+    let intrinsic_gas: u64 = vm.get_intrinsic_gas()?;
+
+    // calldata_cost = tokens_in_calldata * 4
+    let calldata_cost: u64 =
+        gas_cost::tx_calldata(&calldata, vm.env.config.fork).map_err(VMError::OutOfGas)?;
+
+    // same as calculated in gas_used()
+    let tokens_in_calldata: u64 = calldata_cost
+        .checked_div(STANDARD_TOKEN_COST)
+        .ok_or(VMError::Internal(InternalError::DivisionError))?;
+
+    // floor_cost_by_tokens = TX_BASE_COST + TOTAL_COST_FLOOR_PER_TOKEN * tokens_in_calldata
+    let floor_cost_by_tokens = tokens_in_calldata
+        .checked_mul(TOTAL_COST_FLOOR_PER_TOKEN)
+        .ok_or(VMError::Internal(InternalError::GasOverflow))?
+        .checked_add(TX_BASE_COST)
+        .ok_or(VMError::Internal(InternalError::GasOverflow))?;
+
+    let min_gas_limit = max(intrinsic_gas, floor_cost_by_tokens);
+    if vm.current_call_frame()?.gas_limit < min_gas_limit {
+        return Err(VMError::TxValidation(TxValidationError::IntrinsicGasTooLow));
+    }
+
+    Ok(())
+}
+
+pub fn validate_max_fee_per_blob_gas(
+    vm: &mut VM<'_>,
+    tx_max_fee_per_blob_gas: U256,
+) -> Result<(), VMError> {
+    if tx_max_fee_per_blob_gas
+        < get_base_fee_per_blob_gas(vm.env.block_excess_blob_gas, &vm.env.config)?
+    {
+        return Err(VMError::TxValidation(
+            TxValidationError::InsufficientMaxFeePerBlobGas,
+        ));
+    }
+    Ok(())
+}
+
+pub fn validate_init_code_size(vm: &mut VM<'_>) -> Result<(), VMError> {
+    // [EIP-3860] - INITCODE_SIZE_EXCEEDED
+    if vm.current_call_frame()?.calldata.len() > INIT_CODE_MAX_SIZE
+        && vm.env.config.fork >= Fork::Shanghai
+    {
+        return Err(VMError::TxValidation(
+            TxValidationError::InitcodeSizeExceeded,
+        ));
+    }
+    Ok(())
+}
+
+pub fn validate_sufficient_max_fee_per_gas(vm: &mut VM<'_>) -> Result<(), VMError> {
+    if vm.env.tx_max_fee_per_gas.unwrap_or(vm.env.gas_price) < vm.env.base_fee_per_gas {
+        return Err(VMError::TxValidation(
+            TxValidationError::InsufficientMaxFeePerGas,
+        ));
+    }
+    Ok(())
+}
+
+pub fn validate_type_3_tx(vm: &mut VM<'_>) -> Result<(), VMError> {
```

Comment:
Something like `validate_4844_tx` might be easier to read

Reply:
Done in [8941381](https://github.com/lambdaclass/ethrex/pull/2508/commits/89413816760026a9b2e76800fd7e73150a47dfe2).